goldilocks/src/lib.rs

```diff
@@ -276,6 +276,7 @@ fn reduce128(x: u128) -> Goldilocks {
 }
 
 #[inline]
+#[allow(clippy::cast_possible_truncation)]
 fn split(x: u128) -> (u64, u64) {
     (x as u64, (x >> 64) as u64)
```

Comment:
Alternatively, you could use the bytemuck crate to cast `u128` to `[i64; 2]`.

Reply:
I think I'd slightly prefer just keeping this with the `allow` to minimize dependencies, unless bytemuck ends up being useful in other places